api/src/main/java/com/github/retrooper/packetevents/wrapper/play/server/WrapperPlayServerResourcePackSend.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * This file is part of packetevents - https://github.com/retrooper/packetevents
+ * Copyright (C) 2021 retrooper and contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.github.retrooper.packetevents.wrapper.play.server;
+
+import com.github.retrooper.packetevents.event.PacketSendEvent;
+import com.github.retrooper.packetevents.protocol.packettype.PacketType;
+import com.github.retrooper.packetevents.wrapper.PacketWrapper;
+import net.kyori.adventure.text.Component;
+
+public class WrapperPlayServerResourcePackSend extends PacketWrapper<WrapperPlayServerResourcePackSend> {
+    private String url;
+    private String hash;
+    private boolean required;
+    private Component prompt;
+
+    public WrapperPlayServerResourcePackSend(PacketSendEvent event) {
+        super(event);
+    }
+
+    public WrapperPlayServerResourcePackSend(String url, String hash, boolean required, Component prompt) {
+        super(PacketType.Play.Server.RESOURCE_PACK_SEND);
+        this.url = url;
+        this.hash = hash;
+        this.required = required;
+        this.prompt = prompt;
+    }
+
+    @Override
+    public void read() {
+        url = readString();
+        hash = readString();
+        required = readBoolean();
```

Comment:
https://wiki.vg/index.php?title=Protocol&oldid=16681#Resource_Pack_Send

1.16 and below doesn't have the required or prompt

Reply:
Fixed in latest commit, Required/hasPrompt/Prompt will no longer be accessed unless the server is above or equal to version 1.16.